From stable-bounces@linux.kernel.org  Mon Jul  4 00:29:19 2005
From: Alexander Nyberg <alexn@telia.com>
To: Linux Kernel Mailing List <linux-kernel@vger.kernel.org>
Date: Mon, 04 Jul 2005 09:28:35 +0200
Cc: torvalds@osdl.org, stable@kernel.org
Subject: If ACPI doesn't find an irq listed, don't accept 0 as a valid PCI irq.

If ACPI doesn't find an irq listed, don't accept 0 as a valid PCI irq.
That zero just means that nothing else found any irq information either.

Fixes http://bugme.osdl.org/show_bug.cgi?id=4824

Signed-off-by: Chris Wright <chrisw@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/acpi/pci_irq.c |    2 +-
 1 files changed, 1 insertion(+), 1 deletion(-)

--- linux-2.6.12.2.orig/drivers/acpi/pci_irq.c	2005-07-13 10:53:01.000000000 -0700
+++ linux-2.6.12.2/drivers/acpi/pci_irq.c	2005-07-13 10:56:30.000000000 -0700
@@ -433,7 +433,7 @@
 		printk(KERN_WARNING PREFIX "PCI Interrupt %s[%c]: no GSI",
 			pci_name(dev), ('A' + pin));
 		/* Interrupt Line values above 0xF are forbidden */
-		if (dev->irq >= 0 && (dev->irq <= 0xF)) {
+		if (dev->irq > 0 && (dev->irq <= 0xF)) {
 			printk(" - using IRQ %d\n", dev->irq);
 			acpi_register_gsi(dev->irq, ACPI_LEVEL_SENSITIVE, ACPI_ACTIVE_LOW);
 			return_VALUE(0);
